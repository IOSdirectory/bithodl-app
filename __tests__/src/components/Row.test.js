import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Row from '../../../src/components/Row';

const RealDate = Date;

const mockDate = (fakeDate) => {
  global.Date = class extends RealDate {
    constructor(specifiedDate) {
      super(...arguments);
      return new RealDate(specifiedDate || fakeDate);
    }
  };
};

const unmockDate = () => {
  global.Date = RealDate;
};

jest.mock('../../../src/containers/BtcLabelContainer', () => 'BtcLabelContainer');

describe('Row', () => {
  let row;
  let mockItem;

  beforeAll(() => {
    mockDate('29 November, 2017, 21:59 GMT+02:00');
  });

  beforeEach(() => {
    row = new Row();

    mockItem = {
      id: 'cb3cd64d-4663-49a4-81f5-9a9985be58e9',
      hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
      script: 'OP_RETURN',
      amount: 2000000,
      createdAt: 1471651200,
      unlockAt: 1482192000
    };

    row.props = {
      item: mockItem,
      onShowItem: jest.fn()
    };
  });

  afterAll(() => {
    unmockDate();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Row item={mockItem} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('#_onPress()', () => {
    it('invokes props.onShowItem() with props.item as argument', () => {
      row._onPress();
      expect(row.props.onShowItem).toBeCalledWith(row.props.item);
    });
  });
});
