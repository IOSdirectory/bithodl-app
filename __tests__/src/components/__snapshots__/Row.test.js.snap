// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "borderBottomColor": "#F9F8F8",
          "borderBottomWidth": 1,
          "padding": 16,
          "paddingLeft": 64,
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <View
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 16,
              "height": 32,
              "justifyContent": "center",
              "width": 32,
            },
            Object {
              "backgroundColor": "#D9E5C3",
            },
          ],
          Object {
            "left": -48,
            "position": "absolute",
            "top": 3,
            "zIndex": 100,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          }
        }
      >
        <View
          borderWidth={0}
          color="#7FA835"
          progress={1}
          size={32}
          thickness={1}
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Array [
              Object {
                "fontSize": 11,
                "fontWeight": "bold",
              },
              Object {
                "color": "#7FA835",
              },
            ],
            undefined,
          ]
        }
      >
        ✓
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "color": "#504F63",
            "fontSize": 15,
            "fontWeight": "600",
            "lineHeight": 20,
          },
        ]
      }
    >
      Hodled for 4 months
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "color": "#8F8E94",
            "fontSize": 13,
            "fontWeight": "400",
            "lineHeight": 18,
          },
        ]
      }
    >
      Ended
       
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "color": "#8F8E94",
              "fontSize": 13,
              "fontWeight": "400",
              "lineHeight": 18,
            },
          ]
        }
      >
        20 Dec, 2016 01:00
      </Text>
    </Text>
    <BtcLabelContainer
      amount={2000000}
      style={
        Object {
          "color": "#504F63",
          "fontSize": 13,
          "fontWeight": "600",
          "position": "absolute",
          "right": 20,
          "top": 11,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#EBEBEB",
            "fontSize": 22,
            "position": "absolute",
            "right": 0,
            "top": 8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
